manager/src/lib.rs

```diff
@@ -616,7 +602,25 @@ impl CronManager {
 
         // only skip scheduling if user didnt intend
         let current_slot = self.get_slot_id(None);
-        let call_balance_used = self.task_balance_uses(&task);
+        // let call_balance_used = self.task_balance_uses(&task);
+        let call_balance_used = u128::from(env::used_gas());
+        let mut agent = self.agents.get(&env::signer_account_id()).unwrap();
+
+        // Increment agent reward & task count
+        // Reward for agent MUST include the amount of gas used as a reimbursement
+        let call_total_fee = call_balance_used + self.agent_fee;
+        agent.balance = U128::from(agent.balance.0 + call_total_fee);
+        agent.total_tasks_executed = U128::from(agent.total_tasks_executed.0 + 1);
+
+        // Update agent storage
+        self.agents.insert(&env::predecessor_account_id(), &agent);
+
+        // Decrease task balance
+        task.total_deposit = U128::from(task.total_deposit.0 - call_total_fee);
+
+        // Update task storage
+        self.tasks.insert(&task_hash, &task);
+
```

Comment:
It does feel like *right here* is where we can check the `task.total_deposit` (which is now subtracted) and ensure that we still have enough to schedule the next recurring instance.

Reply:
roger that